src/main/daml/Daml/Finance/Common/Date/DayCount.daml

```diff
@@ -32,10 +41,77 @@ calcDcf : DayCountConventionEnum -> Date -> Date -> Decimal
 calcDcf _ f t | f > t = error "fromDate is greater than toDate"
 calcDcf Act360 f t = calcDcfAct360 f t
 calcDcf Act365_Fixed f t = calcDcfAct365Fixed f t
+calcDcf ActAct_ISDA f t = error "This calculation requires the knowledge of the stub period. Please, call `calcPeriodDcf` instead."
+calcDcf ActAct_ISMA f t = error "This calculation requires the knowledge of the stub period. Please, call `calcPeriodDcf` instead."
 calcDcf Basis_30360 f t = calcDcf30360 f t
 calcDcf Basis_30360_ICMA f t = calcDcf30360Icma f t
 calcDcf Basis_30E3360 f t = calcDcf30E360 False f t -- this corresponds to 30E/360 without the maturity date adjustment
-calcDcf Basis_30E360 f t = error "This calculation requires the knowledge of the maturity date of the product. Please, use `calcDcf30E360 instead`."
+calcDcf Basis_30E360 f t = error "This calculation requires the knowledge of the maturity date of the product. Please, use `calcDcf30E360` instead."
+
+-- | Calculate day count fraction for a schedule period
+calcPeriodDcf : DayCountConventionEnum -> SchedulePeriod -> Date -> Frequency -> Decimal
+calcPeriodDcf Act360 p maturityDate frequency = calcDcf Act360 p.adjustedStartDate p.adjustedEndDate
+calcPeriodDcf Act365_Fixed p maturityDate frequency = calcDcf Act365_Fixed p.adjustedStartDate p.adjustedEndDate
```

Comment:
It might be up to the user to decide whether to calculate DCF on adjusted vs unadjusted dates, so I would expose it as a boolean flag.